### PR TITLE
Probe BERT by layer

### PR DESCRIPTION
**Pull request by iftenney**
Probe BERT by layer

- Modifications to the BERT wrapper to handle scalar mixing and layer selection appropriately
- Experiment configs for edge probing, as used in https://arxiv.org/abs/1905.05950
- Fix minor path bugs in edge probing analysis code
- Update edge probing sample analysis notebook showing how to do per-layer analysis and extract mixing scalars

### CODE REVIEW

File: jiant/bert/utils.py [hunk 1]
```diff
@@ -44,6 +44,12 @@ def __init__(self, args, cache_dir=None):
             args.input_module, cache_dir=cache_dir
         )
         self.embeddings_mode = args.bert_embeddings_mode
+        self.num_layers = self.model.config.num_hidden_layers
+        if args.bert_max_layer >= 0:
+            self.max_layer = args.bert_max_layer
+        else:
+            self.max_layer = self.num_layers
+        assert self.max_layer <= self.num_layers
```
Comment (sleepinyourhat) [inline jiant/bert/utils.py]: Move this inside the first conditional clause?

File: probing/get_scalar_mix.py [hunk 2]
```diff
@@ -23,13 +23,13 @@
...
 import logging as log
 
-log.basicConfig(format="%(asctime)s: %(message)s", datefmt="%m/%d %I:%M:%S %p", level=log.INFO)
-
 import pandas as pd
 import torch
 
 from typing import List, Tuple, Iterable, Dict
 
+log.basicConfig(format="%(asctime)s: %(message)s", datefmt="%m/%d %I:%M:%S %p", level=log.INFO)
```
Comment (sleepinyourhat) [inline probing/get_scalar_mix.py]: Why this move?

File: jiant/bert/utils.py (hunk 1 above)
Comment (iftenney) [reply jiant/bert/utils.py]: It's a little safer to enforce globally?

File: probing/get_scalar_mix.py (hunk 2 above)
Comment (iftenney) [reply probing/get_scalar_mix.py]: Formatter was complaining.

**Comment by iftenney** [comment]
Will update readme in a separate PR (there's a bunch of other documentation I need to add). Thanks for the review!

File: jiant/bert/utils.py (hunk 1 above)
Comment (sleepinyourhat) [reply jiant/bert/utils.py]: Up to you, but seems more confusing then helpful. The pattern is `x = y; assert x >= y`.

**Pull request merged**